lib/src/Base/Func/SpecFunc/SpecFunc.cxx

```diff
@@ -745,6 +769,36 @@ Scalar SpecFunc::TriGamma(const Scalar x)
   return value + 0.5 * y + (1. + (.16666666666666667 + (-0.33333333333333333e-1 + (0.23809523809523810e-1 + (-0.33333333333333333e-1 + (0.75757575757575758e-1 + (-.25311355311355311 + (1.1666666666666667 + (-7.0921568627450980 + 54.971177944862155 * y) * y) * y) * y) * y) * y) * y) * y) * y) / z;
 }
 
+/* stirlerr(n) = log(n!) - log( sqrt(2*pi*n)*(n/e)^n )
+*     = log Gamma(n+1) - 1/2 * [log(2*pi) + log(n)] - n*[log(n) - 1]
+*     = log Gamma(n+1) - (n + 1/2) * log(n) + n - log(2*pi)/2 */
+Scalar SpecFunc::stirlerr(const UnsignedInteger n)
+{
+  static const Scalar stirlerrTable[26] =
+  {
+    0.000000000000000000000, 8.10614667953272582e-02, 4.13406959554092941e-02,
+    2.76779256849983391e-02, 2.07906721037650931e-02, 1.66446911898211922e-02,
+    1.38761288230707480e-02, 1.18967099458917701e-02, 1.04112652619720965e-02,
+    9.25546218271273292e-03, 8.33056343336287126e-03, 7.57367548795184079e-03,
+    6.94284010720952987e-03, 6.40899418800420707e-03, 5.95137011275884774e-03,
+    5.55473355196280137e-03, 5.20765591960964044e-03, 4.90139594843473786e-03,
+    4.62915374933402859e-03, 4.38556024923232427e-03, 4.16631969199692246e-03,
+    3.96795421864085962e-03, 3.78761806844443458e-03, 3.62296022468309471e-03,
+    3.47202138297876696e-03, 3.33315563672809288e-03
+  };
+  if (n < 26) return stirlerrTable[n];
+  static Scalar S0 = 8.33333333333333333e-02;
+  static Scalar S1 = 2.77777777777777778e-03;
+  static Scalar S2 = 7.93650793650793651e-04;
+  static Scalar S3 = 5.95238095238095238e-04;
+  static Scalar S4 = 8.41750841750841751e-04;
+  const Scalar nn = (1.0 * n) * n;
+  if (n > 2559) return (S0 - S1 / nn) / n;
+  if (n > 82)  return (S0 - (S1 - S2 / nn) / nn) / n;
+  if (n > 50)  return (S0 - (S1 - (S2 - S3 / nn) / nn) / nn) / n;
+  return (S0 - (S1 - (S2 - (S3 - S4 / nn) / nn) / nn) / nn) / n;
+}
+
```

Comment:
for reference, where did you get this ?

I tried to look elsewhere for comparison, but I did not find the same implementation:
https://kaskr.github.io/adcomp/stirlerr_8cpp_source.html

Reply:
The formula given in Loader's paper does not provide full double precision. I had to increase the threshold a little bit to achieve the needed precision and to increase the lookup table accordingly.